lib/web/web.go

```diff
@@ -0,0 +1,70 @@
+package web
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type Response interface{}
+
+type ServerFunc func(w http.ResponseWriter, r *http.Request) (Response, error)
+
+func ResponseHandler(f ServerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		resp, err := f(w, r)
+		if err != nil {
+			handleError(w, err)
+			return
+		}
+		handleResponse(w, resp)
+	}
+}
+
+type webError struct {
+	Message string `json:"message"`
+}
+
+type errorResponse struct {
+	Error webError `json:"error"`
+}
```

Comment:
Should this and `successResponse` be combined into a single type that can be exported for the SDK's decoding step? We're going to have another naming problem since I already have `type Response interface{}` that's just a placeholder for the handler functions to return. That's really probably `type Data interface{}` or something.